core/lib/generators/solidus/install/install_generator/install_frontend.rb

```diff
@@ -24,22 +24,11 @@ def call(frontend)
 
       private
 
-      def bundle_command(command)
-        @generator_context.say_status :run, "bundle #{command}"
-        bundle_path = Gem.bin_path("bundler", "bundle")
-
-        BundlerContext.bundle_cleanly do
-          system(
-            Gem.ruby,
-            bundle_path,
-            *command.shellsplit,
-          )
-        end
-      end
-
       def install_solidus_frontend
         unless @bundler_context.component_in_gemfile?(:frontend)
-          bundle_command 'add solidus_frontend'
+          # `Rails::Generator::AppBase#bundle_command` is protected so have to `send` it.
+          # See https://api.rubyonrails.org/v3.2.16/classes/Rails/Generators/AppBase.html#method-i-run_bundle
+          @generator_context.send :bundle_command, 'add solidus_frontend'
```

Comment:
@elia @waiting-for-dev Any concerns with using `#send` here?

Reply:
As we own the `@generator_context`, we can wrap `bundle_command` in another method and call it from `InstallFrontend`. However, it adds up to my fears that we're leveraging private API by relying on `AppBase` 😨

Reply:
@waiting-for-dev I'm not sure if I understood you correctly, but if I try to add a `public_bundle_command` method to InstallGenerator, it causes `solidus:install` to fail:

```ruby
module Solidus
  class InstallGenerator < Rails::Generators::AppBase
    def public_bundle_command(command)
      bundle_command(command)
    end
  end
end
```

```sh
solidus 14:19:05 $ bin/sandbox --frontend=solidus_frontend
# ...
/home/gsmendoza/repos/solidusio/solidus/core/lib/generators/solidus/install/install_generator.rb:61:in `public_bundle_command': rails public_bundle_command takes 1 argument, but it should not. (ArgumentError)
```

I think this is because `solidus:install` attempts to run all of the public methods in InstallGenerator, including the `public_bundle_command` helper.

It doesn't seem like there's a way to avoid using `#send` here. Do you have any other ideas?

Reply:
Oh, too bad 😢 Let's go with `send` or revert to what we had before—your choice.

Reply:
@waiting-for-dev Let's go with `send`. We're already using AppBase anyway hehehe...